test/integration/integration_test.go

```diff
@@ -3,244 +3,106 @@
-				t.Logf("failed to clean up object %#v: %s", o, err)
-				anyFailed = true
-			}
-		}
-		if anyFailed {
-			t.Fatalf("failed to clean up resources")
-		}
-	}
-	defer cleanup()
-
-	err = sdk.Create(appNamespace)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(appDeployment)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(appService)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(appRouteDefault)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(appRouteInternal)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(clusterIngressDefault)
-	if err != nil {
-		t.Fatal(err)
-	}
-	err = sdk.Create(clusterIngressInternal)
-	if err != nil {
-		t.Fatal(err)
-	}
+func TestIntegration(t *testing.T) {
+	tc := NewTestConfig(t, *clusterName, *manifestsDir)
 
-	for _, service := range []*corev1.Service{defaultService, internalService} {
-		err := wait.Poll(1*time.Second, 2*time.Minute, func() (bool, error) {
-			err := sdk.Get(service)
-			if err != nil {
-				if errors.IsNotFound(err) {
-					return false, nil
-				}
-				return false, err
-			}
-			for _, ingress := range service.Status.LoadBalancer.Ingress {
-				if len(ingress.IP) > 0 {
-					t.Logf("service %s/%s has ingress.IP %s", service.Namespace, service.Name, ingress.IP)
-					return true, nil
-				}
-			}
-			return false, nil
-		})
-		if err != nil {
-			t.Fatalf("timed out waiting for service %s/%s: %s", service.Namespace, service.Name, err)
-		}
-	}
+	tc.setup(t)
+	defer tc.teardown(t)
 
-	client := &http.Client{}
-	for routeHost, ingressIP := range map[string]string{
-		appRouteDefault.Spec.Host:  defaultService.Status.LoadBalancer.Ingress[0].IP,
-		appRouteInternal.Spec.Host: internalService.Status.LoadBalancer.Ingress[0].IP,
-	} {
-		err := wait.Poll(1*time.Second, 2*time.Minute, func() (bool, error) {
-			req, err := http.NewRequest("GET", "http://"+ingressIP, nil)
-			req.Host = routeHost
-			resp, err := client.Do(req)
-			defer resp.Body.Close()
-			if err != nil {
-				return false, err
-			}
-			if resp.StatusCode == http.StatusOK {
-				return true, nil
-			}
-			return false, fmt.Errorf("last response: %s", resp.Status)
-		})
-		if err != nil {
-			t.Fatalf("timed out waiting for route endpoint %q at ingress IP %q: %s", routeHost, ingressIP, err)
-		}
-	}
+	// Execute subtests
+	t.Run("TestDefaultIngress", func(t *testing.T) { testDefaultIngress(t, tc) })
 }
 
 type TestConfig struct {
 	operatorNamespace string
 	clusterName       string
+	manifestsDir      string
 	kubeConfig        string
-
-	t *testing.T
 }
 
-func NewTestConfig(t *testing.T) *TestConfig {
-	config := &TestConfig{t: t}
-
+func NewTestConfig(t *testing.T, clusterName string, manifestsDir string) *TestConfig {
 	// Check prerequisites
 	kubeConfig := os.Getenv("KUBECONFIG")
 	if len(kubeConfig) == 0 {
 		t.Fatalf("KUBECONFIG is required")
 	}
 	// The operator-sdk uses KUBERNETES_CONFIG...
 	os.Setenv("KUBERNETES_CONFIG", kubeConfig)
```

Comment:
Is this `os.Setenv` still needed now that `test-integration.sh` sets `KUBERNETES_CONFIG`?

Reply:
Gone